src/providers/hana/qgshananewconnection.cpp

```diff
@@ -0,0 +1,348 @@
+    {
+      QMessageBox::warning( this,
+                            tr( "Save Connection" ), tr( "WARNING: Tenant database name cannot be empty." ), QMessageBox::Ok );
+      return;
+    }
+  }
+
+  QString connName = txtName->text();
+  QgsHanaSettings::setSelectedConnection( connName );
+  bool hasAuthConfigID = !mAuthSettings->configId().isEmpty();
+
+  if ( !hasAuthConfigID && mAuthSettings->storePasswordIsChecked() &&
+       QMessageBox::question( this,
+                              tr( "Saving Passwords" ),
+                              tr( "WARNING: You have opted to save your password. It will be stored in unsecured "
+                                  "plain text in your project files and in your home directory (Unix-like OS) or user profile (Windows). "
+                                  "If you want to avoid this, press Cancel and either:\n\na) Don't save a password in the connection "
+                                  "settings  — it will be requested interactively when needed;\nb) Use the Configuration tab to add your "
+                                  "credentials in an HTTP Basic Authentication method and store them in an encrypted database." ),
+                              QMessageBox::Ok | QMessageBox::Cancel ) == QMessageBox::Cancel )
+  {
+    return;
+  }
+
+  QgsHanaSettings settings( connName, true );
+  // warn if entry was renamed to an existing connection
+  if ( ( !mOriginalConnName.isNull() && mOriginalConnName.compare( connName, Qt::CaseInsensitive ) != 0 ) &&
+       QMessageBox::question( this, tr( "Save Connection" ), tr( "Should the existing connection %1 be overwritten?" ).arg( connName ),
+                              QMessageBox::Ok | QMessageBox::Cancel ) == QMessageBox::Cancel )
+  {
+    return;
+  }
+
+  // on rename delete the original entry first
+  if ( !mOriginalConnName.isNull() && mOriginalConnName != connName )
+    QgsHanaSettings::removeConnection( mOriginalConnName );
+
+  readSettingsFromControls( settings );
+  if ( !mAuthSettings->storeUsernameIsChecked() )
+    settings.setUserName( QString( "" ) );
+  if ( !( mAuthSettings->storePasswordIsChecked() && !hasAuthConfigID ) )
+    settings.setPassword( QString( "" ) );
+  settings.setSaveUserName( mAuthSettings->storeUsernameIsChecked() );
+  settings.setSavePassword( mAuthSettings->storePasswordIsChecked() && !hasAuthConfigID );
+
+  settings.save();
+
+  QDialog::accept();
+}
+
+void QgsHanaNewConnection::btnConnect_clicked()
+{
+  testConnection();
+}
+
+void QgsHanaNewConnection::cmbIdentifierType_changed( int index )
+{
+  if ( QgsHanaIdentifierType::fromInt( static_cast<uint>( index ) ) == QgsHanaIdentifierType::INSTANCE_NUMBER )
+  {
+    txtIdentifier->setMaxLength( 2 );
+    txtIdentifier->setValidator( new QIntValidator( 0, 99, this ) );
+    txtIdentifier->setText( QStringLiteral( "00" ) );
+  }
+  else
+  {
+    txtIdentifier->setMaxLength( 5 );
+    txtIdentifier->setValidator( new QIntValidator( 10000, 99999, this ) );
```

Comment:
Port numbers are 16-bit numbers. Therefore, the highest available port is 65535. Why don't we allow port numbers less than 10000?

Reply:
Done in e8995d3.